### ISSUE TITLE
Incorrect layer names in activation dict

### ISSUE THREAD

**Issue by NelsonSharma** [philipperemy/keract#94]
check the attached text file
Hi, input layer names and shapes have been mapped incorrectly in activation dictionary,
 kindly check below, this model has 4 inputs but only 2 seem to be correctly mapped
[problem_mapping_layer_names.txt](https://github.com/philipperemy/keract/files/4683493/problem_mapping_layer_names.txt)

**Comment by philipperemy** [comment]
@NelsonSharma interesting. Thanks for reporting. Do you mind sharing your code so that I can update the lib?

**Comment by NelsonSharma** [comment]
hi, plz check shared code file 
[git_keract_issue.py.txt](https://github.com/philipperemy/keract/files/4686028/git_keract_issue.py.txt)

**Comment by philipperemy** [comment]
I see 4 inputs
```
input_N (InputLayer)            [(None, 384, 1)]     0                                            
__________________________________________________________________________________________________
input_C (InputLayer)            [(None, 384, 1)]     0        
input_N_dur (InputLayer)        [(None, 1)]          0                     
input_C_dur (InputLayer)        [(None, 1)]          0     
```

And 4 activations in your code:
```
**input_N 	 (1, 384, 1)
input_C 	 (1,)
input_N_dur 	 (1, 384, 1)
input_C_dur 	 (1,)**
conv_N_1 	 (1, 378, 30)
conv_C_1 	 (1, 378, 30)
pool_N_2 	 (1, 189, 30)
pool_C_2 	 (1, 189, 30)
conv_N_3 	 (1, 187, 20)
conv_C_3 	 (1, 187, 20)
pool_N_4 	 (1, 93, 20)
pool_C_4 	 (1, 93, 20)
conv_N_5 	 (1, 91, 10)
conv_C_5 	 (1, 91, 10)
pool_N_6 	 (1, 45, 10)
pool_C_6 	 (1, 45, 10)
flat_N_7 	 (1, 450)
flat_C_7 	 (1, 450)
den_Ndur_concat 	 (1, 451)
den_Cdur_concat 	 (1, 451)
den_NC_concat 	 (1, 902)
den_NC_01 	 (1, 20)
den_NC_02 	 (1, 10)
den_out 	 (1, 2)

```

**Comment by NelsonSharma** [comment]
sure, but that's not the issue at all. It's not the number of inputs, but the inputs themselves. See here carefully.

the input layer named 'input_N_dur' has a shape (1,1) in the model.
input_N_dur (InputLayer)        [(None, 1)]    

but in the activations dictionary (one returned by _get_activations_ function) is (384,1)
input_N_dur 	 (1, 384, 1)

As I understood, the layer  'input_C' and 'input_N_dur' have been swapped in activations dictionary

**Comment by philipperemy** [comment]
@NelsonSharma fix on the way. https://github.com/philipperemy/keract/pull/96/files

**Comment by philipperemy** [comment]
I've also added an extra test.

```python
    def test_inputs_order(self):
        i10 = Input(shape=(10,), name='i1')
        i40 = Input(shape=(40,), name='i4')
        i30 = Input(shape=(30,), name='i3')
        i20 = Input(shape=(20,), name='i2')

        a = Dense(1, name='fc1')(concatenate([i10, i40, i30, i20], name='concat'))
        model = Model(inputs=[i40, i30, i20, i10], outputs=[a])
        x = [
            np.random.uniform(size=(1, 40)),
            np.random.uniform(size=(1, 30)),
            np.random.uniform(size=(1, 20)),
            np.random.uniform(size=(1, 10))
        ]

        acts = get_activations(model, x)
        self.assertListEqual(list(acts['i1'].shape), [1, 10])
        self.assertListEqual(list(acts['i2'].shape), [1, 20])
        self.assertListEqual(list(acts['i3'].shape), [1, 30])
        self.assertListEqual(list(acts['i4'].shape), [1, 40])
```

**Comment by philipperemy** [comment]
Merged https://github.com/philipperemy/keract/commit/5ab5a4d0a0630fcea9ce57b420869a8daeeba3b2

**Comment by philipperemy** [comment]
Released in 3.1.1.
